Can't iterate through a slice of channels

The following program `main.go` triggers a panic:

```go
package main

import "fmt"

func main() {
	v := []chan struct{}{make(chan struct{})}

	for _, vv := range v {
		_ = vv
	}

	fmt.Println("success")
}

```

Expected result:
```console
$ go run main.go 
success
```

Got:
```console
$ yaegi main.go
main.go:6:23: panic
reflect.Set: value of type chan struct {} is not assignable to type []chan struct {}
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000380000, 0x40)
	/usr/lib/go/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00019bdf0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:320 +0xbc
panic(0xc98360, 0xc000070790)
	/usr/lib/go/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003580a0, 0xc000281300)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:113 +0x20c
panic(0xc98360, 0xc000070790)
	/usr/lib/go/src/runtime/panic.go:969 +0x166
reflect.Value.assignTo(0xc92460, 0xc00037e000, 0x12, 0xdca660, 0xb, 0xc8a680, 0x0, 0xc92460, 0x0, 0xc00037e000)
	/usr/lib/go/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0xc8a680, 0xc00000e8a0, 0x197, 0xc92460, 0xc00037e000, 0x12)
	/usr/lib/go/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp._make.func3(0xc0003580a0, 0x4b49bd)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:2346 +0xaf
github.com/containous/yaegi/interp.runCfg(0xc000281300, 0xc0003580a0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000001500, 0xc000280600, 0xc000358000)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000001500, 0xc00035e000, 0x94, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jlevesy/code/src/github.com/containous/yaegi/interp/interp.go:388 +0x41f
main.main()
	/home/jlevesy/code/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:138 +0x553
```

--- a/_test/composite10.go
+++ b/_test/composite10.go
@@ -1,0 +1,14 @@
+package main
+
+import "fmt"
+
+func main() {
+	a := []map[int]int{make(map[int]int)}
+
+	for _, b := range a {
+		fmt.Println(b)
+	}
+}
+
+// Output:
+// map[]

--- a/_test/composite9.go
+++ b/_test/composite9.go
@@ -1,0 +1,14 @@
+package main
+
+import "fmt"
+
+func main() {
+	a := [][]int{make([]int,0)}
+
+	for _, b := range a {
+		fmt.Println(b)
+	}
+}
+
+// Output:
+// []

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -248,6 +248,10 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 				case binaryExpr, unaryExpr:
 					// Do not attempt to propagate composite type to operator expressions,
 					// it breaks constant folding.
+				case callExpr:
+					if c.typ, err = nodeType(interp, sc, c); err != nil {
+						return false
+					}
 				default:
 					c.typ = n.typ
 				}

